Utils/test_upload_flow/verify_bucket.py

```diff
@@ -0,0 +1,396 @@
+import argparse
+import functools
+import json
+import os
+from pathlib import Path
+import sys
+import tempfile
+from zipfile import ZipFile
+from packaging.version import Version
+
+from Tests.Marketplace.marketplace_services import init_storage_client
+from Tests.Marketplace.upload_packs import download_and_extract_index
+from Tests.scripts.utils.log_util import install_logging
+from Tests.scripts.utils import logging_wrapper as logging
+
+
+MSG_DICT = {
+    "verify_new_pack": "verified the new pack in the index and that version 1.0.0 zip exists under the pack path",
+    "verify_modified_pack": "verified the packs new version is in the index and that all the new items are present in the pack",
+    "verify_new_version": "verified the new pack's version exists in the index and that the release notes is parsed correctly "
+                          "in the changelog",
+    "verify_rn": "verified the content of the release notes is in the changelog under the right version",
+    "verify_hidden": "verified the pack does not exist in index",
+    "verify_readme": "verified the readme content is parsed correctly and that there was no version bump "
+                     "if only readme was modified",
+    "verify_failed_pack": "verified the commit hash is not updated in the pack metadata in the index.zip",
+    "verify_modified_item_path": "verified the path of the pack item is modified",
+    "verify_dependency": "verified the new dependency is in the pack metadata",
+    "verify_new_image": "verified the new image was uploaded"
+}
+XSOAR_BUCKET = "marketplace-dist-dev"
+XSIAM_BUCKET = "marketplace-v2-dist-dev"
+
+
+def read_json(path):
+    with open(path, 'r') as file:
+        return json.load(file)
+
+
+def logger(func):
+    @functools.wraps(func)
+    def wrapper(self, *args, **kwargs):
+
+        logging.info(f"Starting validation - {func.__name__} for pack '{args[0]}'")
+        try:
+            result, pack_id = func(self, *args, **kwargs)
+            self.is_valid = self.is_valid and result
```

Comment:
it took me long time to understand that self.is_valid is updated here - let's make sure it's documented in GCP class (e.g. in  the init method)

Reply:
done